src/erasure-code/shec/CMakeLists.txt

```diff
@@ -33,4 +33,72 @@ add_dependencies(ec_shec_generic ${CMAKE_SOURCE_DIR}/src/ceph_ver.h)
 set_target_properties(ec_shec_generic PROPERTIES VERSION 1.0.0 SOVERSION 1)
 install(TARGETS ec_shec_generic DESTINATION lib/erasure-code)
 
-#TODO:build libec_shec_neon, libec_shec+sse3, libec_shec_sse4 libraries
+# ec_shec_sse3
+set(shec_neon_srcs
+  erasure-code/jerasure/gf-complete/src/neon/gf_w4_neon.c 
+  erasure-code/jerasure/gf-complete/src/neon/gf_w8_neon.c 
+  erasure-code/jerasure/gf-complete/src/neon/gf_w16_neon.c 
+  erasure-code/jerasure/gf-complete/src/neon/gf_w32_neon.c 
+  erasure-code/jerasure/gf-complete/src/neon/gf_w64_neon.c)
+
+#TODO:build libec_shec_neon library
+if(ARM_NEON OR ARM_NEON2)
+  if(ARM_NEON)
+    set(ARM_NEON_FLAGS "-mfpu=neon")
+  else(ARM_NEON)
+    set(ARM_NEON_FLAGS "-march=armv8-a+simd")
+  endif(ARM_NEON)
+
+  add_library(ec_shec_neon SHARED ${shec_neon_srcs})
+  add_dependencies(ec_shec_neon ${CMAKE_SOURCE_DIR}/src/ceph_ver.h)
+  target_link_libraries(ec_shec_neon ${EXTRALIBS})
+  set_target_properties(ec_shec_neon PROPERTIES VERSION 2.0.0 SOVERSION 2
+    COMPILE_FLAGS ${ARM_NEON_FLAGS})
+  install(TARGETS ec_shec_neon DESTINATION lib/erasure-code)
+else(ARM_NEOR OR ARM_NEON2)
+  message(STATUS "Skipping target ec_shec_neon: Architecture not ARM")
+endif(ARM_NEON OR ARM_NEON2)
+
+# ec_shec_sse3
+if(INTEL_SSE)
+  set(SHEC_SSE3_FLAGS "-msse")
```

Comment:
all of this logic for SHEC_SSE3_FLAGS is duplicated for JERASURE_SSE3_FLAGS. can we move both of these up to erasure-code/CMakeLists.txt with the rest of the try_compile() stuff? i.e. construct variables for INTEL_SSE3_FLAGS and INTEL_SSE4_FLAGS, and use them for both shec and jerasure